src/base/AnyMap.cpp

```diff
@@ -189,7 +189,14 @@ struct convert<Cantera::AnyValue> {
             // Scalar nodes are int, doubles, or strings
             std::string nodestr = node.as<std::string>();
             if (isInt(nodestr)) {
-                target = intValue(nodestr);
+                try {
+                    target = node.as<long int>();
+                } catch (YAML::BadConversion&) {
+                    // This exception is raised if the value doesn't fit in a
+                    // long int, in which case we would rather store it
+                    // (possibly inexactly) as a double.
+                    target = node.as<double>();
+                }
```

Comment:
In `ctml2yaml.py`, I had to add some code to remove integer-like reaction `id` fields (e.g., `0001`), because they raised an error in the YAML somewhere (sorry, I don't recall the exact message). Does this fix that problem, and can that code be removed from `ctml2yaml.py`?

Reply:
I don't think this fixes that problem. Some resolution to that is on my list of rough edges in the YAML format that need to be fixed.